src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,37 @@
+package duke.tasks;
+
+import duke.exceptions.InvalidTimeStampException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import java.time.format.DateTimeParseException;
+
+/** Class representing a deadline */
+public class Deadline extends Task {
+
+    private static DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("d/M/yyyy' 'HHmm");
+    private static DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("MMM d yyyy, h:mma");
+    private LocalDateTime timeInfo;
+
+    public Deadline(String taskDetails, String timeInfoString) throws InvalidTimeStampException {
+        super(taskDetails);
+        try {
+            this.timeInfo = LocalDateTime.parse(timeInfoString, inputFormatter);
+        } catch (DateTimeParseException e) {
+            throw new InvalidTimeStampException(timeInfoString);
+        }
+    }
+
+    @Override
+    public String toString() {
+        String timeInfoString = timeInfo.format(outputFormatter);
```

Comment:
I think just leaving at timeInfo will do since it's declared as String already and there are no other names in the method requiring the same name!